### PR TITLE
OI-2848: Add donate modal to datasets page

### PR DESCRIPTION
**Pull request by moz-rotimib**

### CODE REVIEW

File: web/src/components/pages/datasets/dataset-download-email-prompt/test-utils.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import {
+  ByRoleMatcher,
+  ByRoleOptions,
+  Matcher,
+  SelectorMatcherOptions,
+  fireEvent,
+} from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+
+type downloadDatasetParams = {
```
Comment (moz-dfeller) [inline web/src/components/pages/datasets/dataset-download-email-prompt/test-utils.ts:10]: Types should start with a capital letter
```suggestion
type DownloadDatasetParams = {
```

File: web/src/components/pages/datasets/dataset-download-email-prompt/test-utils.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  ) => HTMLElement
+}
+
+export const downloadDataset = ({
+  queryByRole,
+  getByLabelText,
+  getByRole,
+  isSubscribedToMailingList = false,
+  queryByLabelText,
+}: downloadDatasetParams) => {
```
Comment (moz-dfeller) [inline web/src/components/pages/datasets/dataset-download-email-prompt/test-utils.ts:27]:
```suggestion
}: DownloadDatasetParams) => {
```

**Review by moz-dfeller — CHANGES_REQUESTED** [review]
Small consistency nit picks, else it's all good

File: web/locales/en/messages.ftl
[the remark this answers is not in the document]
Comment (moz-rotimib) [reply web/locales/en/messages.ftl:1]: Yeah I'll remove these automated Pontoon updates now

**Pull request merged**